brayns/common/scene/Scene.h

```diff
@@ -30,10 +30,37 @@
 
 #include <shared_mutex>
 
+SERIALIZATION_ACCESS(ClipPlane)
 SERIALIZATION_ACCESS(Scene)
 
 namespace brayns
 {
+
+class ClipPlane : public BaseObject
+{
+public:
+    ClipPlane(const Plane& plane) : _id(_nextID++), _plane(plane) {}
+
+    size_t getID() const { return _id; }
+
+    const Plane& getPlane() const { return _plane; };
+    void setPlane(const Plane& plane)
+    {
+        _updateValue(_plane, plane);
+    }
+
+    /** @internal */
+    ClipPlane() {}
+    /** @internal */
+    ClipPlane(const size_t id, const Plane& plane) : _id(id), _plane(plane) {}
+
+protected:
+    static size_t _nextID;
```

Comment:
Should this be atomic for threading safety?

Reply:
I discussed it with @tribal-tec, it should be safe as only the main thread creates ClipPlanes